test/test_source_rasterio.py

```diff
@@ -0,0 +1,604 @@
+    image = source.getTile(89, 207, 9)
+    _assertImageMatches(image, 'geotiff_9_89_207')
+
+
+def testTileStyleFromGeotiffs():
+    
+    testDir = os.path.dirname(os.path.realpath(__file__))
+    imagePath = os.path.join(testDir, 'test_files', 'rgb_geotiff.tiff')
+    style = {'band': 1, 'min': 0, 'max': 100,
+             'scheme': 'discrete',
+             'palette': 'matplotlib.Plasma_6'}
+    source = large_image_source_rasterio.open(
+        imagePath, projection='EPSG:3857', style=style)
+    image = source.getTile(22, 51, 7, encoding='PNG')
+    _assertImageMatches(image, 'geotiff_style_7_22_51')
+
+
+def testTileLinearStyleFromGeotiffs():
+    
+    testDir = os.path.dirname(os.path.realpath(__file__))
+    imagePath = os.path.join(testDir, 'test_files', 'rgb_geotiff.tiff')
+    style = {'band': 1, 'min': 0, 'max': 100,
+             'palette': 'matplotlib.Plasma_6',
+             'scheme': 'linear'}
+    source = large_image_source_rasterio.open(
+        imagePath, projection='EPSG:3857', style=style, encoding='PNG')
+    image = source.getTile(22, 51, 7)
+    _assertImageMatches(image, 'geotiff_style_linear_7_22_51')
+
+
+def testTileStyleBadInput():
+    
+    def _assertStyleResponse(imagePath, style, message):
+        
+        with pytest.raises((TileSourceError, ValueError), match=message):
+            source = large_image_source_rasterio.open(
+                imagePath, projection='EPSG:3857', style=style, encoding='PNG')
+            source.getTile(22, 51, 7)
+
+    testDir = os.path.dirname(os.path.realpath(__file__))
+    imagePath = os.path.join(testDir, 'test_files', 'rgb_geotiff.tiff')
+
+    _assertStyleResponse(imagePath, {
+        'band': 1.1,
+    }, 'Band has to be a positive integer, -1, or a band interpretation found in the source.')
+
+    _assertStyleResponse(imagePath, {
+        'band': 500,
+    }, 'Band has to be a positive integer, -1, or a band interpretation found in the source.')
+
+    _assertStyleResponse(imagePath, {
+        'band': 1,
+        'palette': 'nonexistent.palette'
+    }, 'cannot be used as a color palette')
+
+    _assertStyleResponse(imagePath, ['style'], 'Style is not a valid json object.')
+
+
+def testThumbnailFromGeotiffs():
+    
+    testDir = os.path.dirname(os.path.realpath(__file__))
+    imagePath = os.path.join(testDir, 'test_files', 'rgb_geotiff.tiff')
+    source = large_image_source_rasterio.open(imagePath)
+    
+    # We get a thumbnail without a projection
+    image, mimeType = source.getThumbnail(encoding='PNG')
+    assert image[:len(utilities.PNGHeader)] == utilities.PNGHeader
+    
+    # We get a different thumbnail with a projection
+    source = large_image_source_rasterio.open(imagePath, projection='EPSG:3857')
+    image2, mimeType = source.getThumbnail(encoding='PNG')
+    assert image2[:len(utilities.PNGHeader)] == utilities.PNGHeader
+    assert image != image2
+
+
+def testPixel():
```

Comment:
@manthey, I think I'll need some support satrting from here, I cannot manage to make the test to work and it's coming from outside of the actual source file so I think I'll loose my time trying to understand what's wrong. 

When I run the test for test/test_source_rasterio.py::testPixel, I always get an empty rgb values. could you have a look ?

Reply:
I'll take a look.

Reply:
If you open a sample file (e.g., test/test_files/rgb_geotiff.tiff) without a projection and then call getThumbnail, you'll see that returns an all black result.  getPixel is just the victim of this...  I haven't looked into what is happening without a projection yet.

Reply:
hmmm,I need to check then, because when I was looking at the generated jpeg it seemed to me that it was working